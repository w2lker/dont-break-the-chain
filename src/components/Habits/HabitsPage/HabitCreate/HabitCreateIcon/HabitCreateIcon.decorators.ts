import withStyles from '@material-ui/core/styles/withStyles';
import HabitCreateIcon from './HabitCreateIcon';
import habitCreateIconStyles from './HabitCreateIcon.styles';

const styledComponent = withStyles(habitCreateIconStyles)(HabitCreateIcon);

export default styledComponent;
