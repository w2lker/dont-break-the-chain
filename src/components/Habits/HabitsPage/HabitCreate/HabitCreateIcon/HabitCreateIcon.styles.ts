const habitCreateIconStyles = {
  root: {
    padding: 0,
  },
};

export default habitCreateIconStyles;
