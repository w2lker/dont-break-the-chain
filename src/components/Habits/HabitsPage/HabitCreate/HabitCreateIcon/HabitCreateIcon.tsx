import React from 'react';

import { WithStyles } from '@material-ui/styles';

import habitCreateIconStyles from './HabitCreateIcon.styles';

export interface IHabitCreateIconProps extends WithStyles<typeof habitCreateIconStyles> {}

const HabitCreateIcon: React.FC<IHabitCreateIconProps> = (props) => {
  const { classes } = props;
  return (
    <div className={classes.root}>
      HabitCreateIcon Component
    </div>
  );
};

export default HabitCreateIcon;
