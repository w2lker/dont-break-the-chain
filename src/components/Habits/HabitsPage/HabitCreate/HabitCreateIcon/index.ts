import HabitCreateIcon from './HabitCreateIcon.decorators';

export default HabitCreateIcon;
