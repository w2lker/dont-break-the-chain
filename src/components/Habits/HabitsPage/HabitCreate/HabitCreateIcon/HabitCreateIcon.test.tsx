import React from 'react';

import { mount, shallow } from 'enzyme';
// import cases from 'jest-in-case';

import HabitCreateIcon, { IHabitCreateIconProps } from './HabitCreateIcon';
import HabitCreateIconDecorated from './HabitCreateIcon.decorators';
import habitCreateIconStyles from './HabitCreateIcon.styles';

const testSetup = () => {
  const classes = {
    root: 'habitCreateIcon-wrapper',
  };

  const classesKeys = Object.keys(classes);

  const emptyProps: IHabitCreateIconProps = {
    classes,
  };

  const sampleProps: IHabitCreateIconProps = {
    classes,
  };

  return { classes, classesKeys, emptyProps, sampleProps };
};

describe('HabitCreateIcon component', () => {
  it('match snapshot', () => {
    const { sampleProps } = testSetup();
    const component = shallow(<HabitCreateIcon {...sampleProps} />);
    expect(component.debug()).toMatchSnapshot();
  });

  test.todo('sample placeholder');
});

describe('HabitCreateIcon styles', () => {
  it('contains principle classes', () => {
    const { classesKeys } = testSetup();
    expect(habitCreateIconStyles).toBeDefined();
    classesKeys.forEach((keyValue) => {
      // @ts-ignore
      expect(habitCreateIconStyles[keyValue]).toBeDefined();
    });
  });
});

describe('HabitCreateIcon decorators', () => {
  it('provides styled classes from decorators', () => {
    const { sampleProps, classesKeys } = testSetup();
    const component = mount(<HabitCreateIconDecorated {...sampleProps} />);
    // @ts-ignore
    const assignedClasses = component.find('HabitCreateIcon').props().classes;
    classesKeys.forEach(keyValue => {
      expect(assignedClasses[keyValue]).toBeDefined();
    });
  });
});
