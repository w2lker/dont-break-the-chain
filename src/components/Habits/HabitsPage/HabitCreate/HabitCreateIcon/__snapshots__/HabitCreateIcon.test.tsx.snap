// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HabitCreateIcon component match snapshot 1`] = `
"<div className=\\"habitCreateIcon-wrapper\\">
  HabitCreateIcon Component
</div>"
`;
